features/126-record-id/implementation.md

```diff
@@ -0,0 +1,210 @@
+# Removing `type/` from APIs
+
+## Record Payload
+
+```json
+// Old
+{
+	"_id": "note/4d4a6018-d365-4103-b35a-249b972ae614",
+	"_access": null,
+	"category": {
+        "$id": "category/de3365c0-b89e-4e1e-8645-bd1ef33012af",
+        "$type": "ref"
+    },
+}
+
+// New
+{
+	"_recordID": "4d4a6018-d365-4103-b35a-249b972ae614",
+	"_recordType": "note",
+	"_access": null,
+	"category": {
+        "$recordType": "category", 
+        "$recordID": "de3365c0-b89e-4e1e-8645-bd1ef33012af", 
+        "$type": "ref"
+    },
+}
+
+// Transitional (maintaining backward compatibility)
+{
+	"_id": "note/4d4a6018-d365-4103-b35a-249b972ae614",
+	"_recordID": "4d4a6018-d365-4103-b35a-249b972ae614",
+	"_recordType": "note",
+	"_access": null,
+	"category": {
+        "$id": "category/de3365c0-b89e-4e1e-8645-bd1ef33012af",
+        "$recordType": "category", 
+        "$recordID": "de3365c0-b89e-4e1e-8645-bd1ef33012af", 
+        "$type": "ref"
+    },
+}
+```
+
+## Server
+
+```
+// Deprecated
+// Note: There is no plan to update record:fetch because SDK should use record:query.
+{
+    "action": "record:fetch",
+    "ids": ["note/4d4a6018-d365-4103-b35a-249b972ae614"]
+}
+
+// Old
+{
+    "action": "record:delete",
+    "ids": ["note/4d4a6018-d365-4103-b35a-249b972ae614"]
+}
+
+// New
+{
+    "action": "record:delete",
+    "ids": ["4d4a6018-d365-4103-b35a-249b972ae614"]
+    "type": "note"
+}
+```
+
+## iOS SDK
+
+This illustrates how the convenient methods will change. The SKYOperation APIs
+will be updated as well but not illustrated here.
+
+The Record fetch and delete APIs are changed substantially:
+
+```obj-c
+// Old
+- (void)fetchRecordWithID:(SKYRecordID *)recordID
+        completionHandler:(void (^)(SKYRecord *record, NSError *error))completionHandler;
+
+// New
+- (void)fetchRecordWithID:(NSString *)recordID
```

Comment:
Is there any reason to change SKYRecordID to string? (Just asking, no thought from me)

Reply:
My idea is that using string ID is easy to understand for most people. For new learner, they don’t need to figure out how to construct a `SKYRecordID`.

I am open to suggestion for change here, as having `SKYRecordID` might help type safety.